lib/vsc/utils/run.py

```diff
@@ -86,6 +86,48 @@
 SHELL = BASH
 
 
+class CmdList(list):
+    """Wrapper for 'list' type to be used for constructing a list of options & arguments for a command."""
+
+    def __init__(self, cmd=None):
+        """
+        Create CmdList instance to construct command
+
+        :param cmd: actual command to run (first item in list)
+        """
+        super(CmdList, self).__init__()
+        if cmd is not None:
+            if isinstance(cmd, list):
+                super(CmdList, self).extend(cmd)
+            else:
+                super(CmdList, self).append(cmd)
+
+    def add_opts_args(self, items, tmpl_vals=None, allow_spaces=True):
```

Comment:
why not make this append?

Reply:
i) to capture cases where `.append` and `.extend` is being used
ii) `add_opts_args` is a joint alternative for `append`/`extend`

Reply:
I wanted to avoid confusion with `set.add`, but fine, will rename to `.add`